proxy/handlers_test.go

```diff
@@ -260,3 +262,79 @@ func TestProxy_registerDashboardHandlers_jwtEndpoint(t *testing.T) {
 		assert.Equal(t, rawJWT, string(b))
 	})
 }
+
+func TestLoadSessionState(t *testing.T) {
+	t.Parallel()
+
+	t.Run("no session", func(t *testing.T) {
+		t.Parallel()
+
+		opts := testOptions(t)
+		proxy, err := New(&config.Config{Options: opts})
+		require.NoError(t, err)
+
+		r := httptest.NewRequest(http.MethodGet, "/.pomerium/", nil)
+		w := httptest.NewRecorder()
+		proxy.ServeHTTP(w, r)
+
+		assert.Equal(t, http.StatusOK, w.Code)
+		assert.Contains(t, w.Body.String(), "window.POMERIUM_DATA")
+		assert.NotContains(t, w.Body.String(), "___SESSION_ID___")
+	})
+	t.Run("cookie session", func(t *testing.T) {
+		t.Parallel()
+
+		opts := testOptions(t)
+		proxy, err := New(&config.Config{Options: opts})
+		require.NoError(t, err)
+
+		sharedKey, err := opts.GetSharedKey()
+		require.NoError(t, err)
+
+		encoder, err := jws.NewHS256Signer(sharedKey)
+		require.NoError(t, err)
+
+		sessionBS, err := encoder.Marshal(&sessions.State{
+			ID: "___SESSION_ID___",
+		})
+		require.NoError(t, err)
+
+		r := httptest.NewRequest(http.MethodGet, "/.pomerium/", nil)
+		r.AddCookie(&http.Cookie{
+			Name:   opts.CookieName,
+			Domain: opts.CookieDomain,
+			Value:  string(sessionBS),
+		})
+		w := httptest.NewRecorder()
+		proxy.ServeHTTP(w, r)
+
+		assert.Equal(t, http.StatusOK, w.Code)
+		assert.Contains(t, w.Body.String(), "___SESSION_ID___")
+	})
+	t.Run("header session", func(t *testing.T) {
+		t.Parallel()
+
+		opts := testOptions(t)
+		proxy, err := New(&config.Config{Options: opts})
+		require.NoError(t, err)
+
+		sharedKey, err := opts.GetSharedKey()
+		require.NoError(t, err)
+
+		encoder, err := jws.NewHS256Signer(sharedKey)
+		require.NoError(t, err)
+
+		sessionBS, err := encoder.Marshal(&sessions.State{
+			ID: "___SESSION_ID___",
+		})
+		require.NoError(t, err)
```

Comment:
(totally optional)

Might be nice to extract this session token setup code out of the individual "cookie session" / "header session" tests to make it super clear that the same exact token can appear either in a cookie or in the Authorization header.